asf_core_data/pipeline/preprocessing/preprocess_epc_data.py

```diff
@@ -187,28 +195,61 @@ def load_and_preprocess_epc_data(
     # Do not save/overwrite the preprocessed data when not loading entire GB dataset
     # in order to prevent confusion.
 
+    if str(save_data) == "S3":
+        save_data = None
+    else:
+        raw_data_path = data_batches.get_batch_path(
+            base_config.RAW_EPC_DATA_PATH, data_path, batch, check_folder="input"
+        )
+
+        # Raw EPC can be found in data path?
+        raw_epc_exists = (Path(data_path) / raw_data_path).is_file()
+
     if n_samples is not None:
 
         save_data = None
         logging.warning(
             "You're not loading all samples so the processed data will not be saved!"
         )
 
-    if save_data or (save_data is not None and not os.path.isabs(save_data)):
+    if (save_data or (save_data is not None and not os.path.isabs(save_data))) and (
+        save_data is not None
+    ):
```

Comment:
not sure I understand this logic as `save_data` is not a `boolean`.
```suggestion
    if save_data is not None and not os.path.isabs(save_data):
```

Reply:
Oh dear, I think I adjusted this if statement too many times. Yes, your suggestion is sufficient.